polaris-service/src/main/java/org/apache/polaris/service/ratelimiter/OpenTelemetryRateLimiter.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.service.ratelimiter;
+
+/**
+ * Wrapper around the opentelemetry RateLimiter that implements the Polaris RateLimiter interface
+ */
+public class OpenTelemetryRateLimiter implements RateLimiter {
+  io.opentelemetry.sdk.internal.RateLimiter rateLimiter;
+
+  public OpenTelemetryRateLimiter(double creditsPerSecond, double maxBalance, Clock clock) {
+    rateLimiter =
+        new io.opentelemetry.sdk.internal.RateLimiter(
+            creditsPerSecond, maxBalance, new OpenTelemetryClock(clock));
+  }
+
+  @Override
+  public boolean trySpend(double credits) {
+    return rateLimiter.trySpend(credits);
```

Comment:
The javadoc in OpenTelemetry's `trySpend` says:
```
  /**
   * Check to see if the provided cost can be spent within the current limits. Will deduct the cost
   * from the current balance if it can be spent.
   */
```

It seems like we inherit those semantics, but the other RL implementations don't seem to have a concept of "credits" or "balance". In light of that the concept of "spending" is a little unclear. Should those also have a "budget" and "income" (?)

Reply:
I updated the interface to just have `tryAcquire()` to decouple from this. Do you see this conflicting with other rate limiter semantics? I just picked the opentelemetry RateLimiter because it was really easy to understand the code and confirm that it was a reasonable-enough token bucket implementation.

Reply:
For now I think it's okay to have `tryAcquire()`. We can always add a `tryAcquire(int tokens)` or something down the road if we ever need richer semantics like those offered by the OpenTelemetry implementation